### PR TITLE
Fix invalid user info processing

### PR DESCRIPTION
**Pull request by Maximys**
Fix #88265

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System/Uri.MethodsTests.cs [hunk 1]
```diff
@@ -196,6 +196,10 @@ public static IEnumerable<object[]> IsBaseOfTestData()
             yield return new object[] { new Uri("file://C:/path/path/file"), new Uri("file://D:/path/path/path"), false };
             yield return new object[] { new Uri("file://C:/path/path/file"), new Uri("http://host/path/path/file"), false };
             yield return new object[] { new Uri("file://C:/path/path/file"), new Uri("path/path/file", UriKind.Relative), true };
+
+            yield return new object[] { new Uri("https://user@domain.com"), new Uri("https://domain.com"), true };
+            yield return new object[] { new Uri("https://user@domain.com"), new Uri("http://domain.com"), false };
```
Comment (MihaZupan) [inline src/libraries/System.Runtime/tests/System/Uri.MethodsTests.cs:201]:
```suggestion
            yield return new object[] { new Uri("https://user@domain.com"), new Uri("https://domain.com"), true };
            yield return new object[] { new Uri("https://user@domain.com"), new Uri("http://domain.com"), false };
            yield return new object[] { new Uri("https://domain.com"), new Uri("https://user@domain.com"), true };
```

File: src/libraries/System.Private.Uri/src/System/UriExt.cs [hunk 2]
```diff
@@ -866,6 +866,11 @@ public bool IsBaseOf(Uri uri)
 
         internal bool IsBaseOfHelper(Uri uriLink)
         {
+            const UriComponents componentsForCompare =
```
Comment (MihaZupan) [inline src/libraries/System.Private.Uri/src/System/UriExt.cs:869]:
```suggestion
            const UriComponents ComponentsToCompare =
```

**Review by MihaZupan — COMMENTED** [review]
Thanks

File: src/libraries/System.Runtime/tests/System/Uri.MethodsTests.cs (hunk 1 above)
Comment (Maximys) [reply src/libraries/System.Runtime/tests/System/Uri.MethodsTests.cs:201]: Fixed by https://github.com/dotnet/runtime/pull/91106/commits/cf6a59f1c1682918a93174d9d244544894b936ff

File: src/libraries/System.Private.Uri/src/System/UriExt.cs (hunk 2 above)
Comment (Maximys) [reply src/libraries/System.Private.Uri/src/System/UriExt.cs]: Fixed by https://github.com/dotnet/runtime/pull/91106/commits/04a593175e4e1c7ae4591745a899a80741db7451

**Review by MihaZupan — APPROVED** [review]

**Pull request merged**